[ENHANCEMENT] Add related Kotlin project to README

IvanMwiruki opened the pull request:
## Description
The title says it all - this change updates the readme and adds a 30-seconds-of-kotlin link. 

## PR Type
- [ ] Snippets, Tests & Tags (new snippets, updated snippets, re-tagging of snippets, added/updated tests)
- [ ] Scripts & Website & Meta (anything related to files in the [scripts folder](https://github.com/30-seconds/30-seconds-of-code/tree/master/scripts), how the repository's automated procedures work and the website)
- [ ] Glossary & Secondary Features (anything related to the glossary, such as new or updated terms or other secondary features)
- [x] General, Typos, Misc. & Meta (everything else, typos, general stuff and meta files in the repository - e.g. the issue template)

## Guidelines
- [x] I have read the guidelines in the [CONTRIBUTING](https://github.com/30-seconds/30-seconds-of-code/blob/master/CONTRIBUTING.md) document.

fejes713 commented:
@IvanMwiruki very nice work! Congratulations on this collection 🍾 

I pretty much agree with @Chalarangelo on every point. Stuff that he mentioned is only going to do good to your repository and improve its quality. 

Speaking of stars, I don't think this is an issue since the quality of the project is good. 

I give this a green light 🚦 as a _unofficial_ 30-seconds project.

Chalarangelo commented:
@IvanMwiruki I'm gonna merge this as it's now consistent with contribution guidelines. If you need a hand with maintaining your repo or just some guidelines/ideas/anything really, feel free to contact me or any of the other maintainers to help you out!

IvanMwiruki commented:
Thanks for merging in my change. I was in the process of typing out a reply, which is what I have below. 

---

Thanks @fejes713 and @Chalarangelo for the kind words. I really appreciate it. As to @Chalarangelo 's points:

- Done. 
- Thanks for the suggestion. I moved the `Note: This project is inspired by, but in no way affiliated with, 30 Seconds of Code.` to above the Table of Contents and made `Note` bold. 
- I could definitely use more examples. I have a few now but not all snippets contain one. I made the subjective decision of only adding examples to snippets I thought needed to be illustrated. Some snippets seemed simple enough that the description was good enough. But you're right, I'm targeting beginners so I will add more examples. 
- I disagree with this. I like Uncle Bob's idea that instead of adding a comment to code, structure the code so that it doesn't need a comment. I wanted to shoot for code that has a description but would otherwise speak for itself. It seems like even though you're new to Kotlin, you understood what that snippet was doing by reading the description and code. I think there's more value in someone reading the code than reading a description of what the code does. 
- It seems like you have much more experience than I do in this area. So I will take your word for it that automating the README creation will save me in the long run. When I started, I was shooting for the "simplest thing that could possibly work" until I ran into any issues that required a more complex approach. Thanks for the advice. I will look into it. 

I added @Chalarangelo as a collaborator. 

And regarding stars, I posted the project on the internet right after opening this pull request. I now have about 50 stars 😊
